internal/manifests/collector/horizontalpodautoscaler.go

```diff
@@ -25,7 +25,7 @@ import (
 	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
 )
 
-func HorizontalPodAutoscaler(params manifests.Params) client.Object {
+func HorizontalPodAutoscaler(params manifests.Params) (client.Object, error) {
```

Comment:
idea: given this doesn't really return an error, we can remove the error from the signature for now.